### PR TITLE
automatically detect openbsd as platform=linuxbsd

### PR DESCRIPTION
**Pull request by omar-polo**
as per title, like the bit for linux, this automatically matches openbsd as platform=linuxbsd

### CODE REVIEW

**Comment by Calinou** [comment]
There is already a way to alias platforms on line 214:

```python
if selected_platform in ["linux", "bsd", "x11"]:
```

File: SConstruct [hunk 1]
```diff
@@ -196,7 +196,7 @@ elif env_base["p"] != "":
     selected_platform = env_base["p"]
 else:
     # Missing `platform` argument, try to detect platform automatically
-    if sys.platform.startswith("linux"):
+    if sys.platform.startswith("linux") or sys.platform.startswith("openbsd"):
```
Comment (Calinou) [inline SConstruct:199]: Could we also detect other BSDs such as FreeBSD while we're at it?

Comment (omar-polo) [reply SConstruct]: Sure :)

I checked on my server and python recognizes freebsd 1.3 as "freebsd13".  I've added also NetBSD and DragonFlyBSD while at it but I don't have access to those system.

**Comment by Calinou** [comment]
Looks good to me, please squash the commits together so we can merge this. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for instructions :slightly_smiling_face:

**Comment by omar-polo** [comment]
Done :)

**Review by Calinou — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.3.

**Pull request merged**